pkg/routeagent_driver/handlers/calico/ippool_handler.go

```diff
@@ -193,3 +207,79 @@ func (h *calicoIPPoolHandler) deleteIPPool(endpoint *submV1.Endpoint) error {
 func getEndpointSubnetIPPoolName(endpoint *submV1.Endpoint, subnet string) string {
 	return fmt.Sprintf("submariner-%s-%s", endpoint.Spec.ClusterID, strings.ReplaceAll(subnet, "/", "-"))
 }
+
+func (h *calicoIPPoolHandler) platformIsROKS() bool {
+	// Submariner GW is deployed on ROKS using LB service with specific annotations.
+	service, err := h.k8sClient.CoreV1().Services(h.namespace).Get(context.TODO(), gwLBSvcName, metav1.GetOptions{})
+	if err != nil {
+		return false
+	}
```

Comment:
We should check for an unexpected error here, ie other than `IsNotFound`, and return it.

```suggestion
        if apierrors.IsNotFound(err) {
            return false, nil
        }
        
	if err != nil {
		return false, err
	}
```